### PR TITLE
Move subresources to controller.devfile.io APIGroup

### PR DESCRIPTION
**Pull request by amisevsk**
### What does this PR do?
Moves Component and WorkspaceRouting subresources from `workspace.che.eclipse.org` API group to `controller.devfile.io`.

Commits should be broken down into a step-by-step process, with separate commits for autogenerated files.

This PR also removes CSV/OLM related files, since they were already out of date and will be updated separately.

### What issues does this PR fix or reference?
Part of https://github.com/eclipse/che/issues/17122

### Is it tested? How?
Tested on `crc`.

### CODE REVIEW

File: pkg/controller/workspacerouting/solvers/openshift_oauth_solver.go [hunk 1]
```diff
@@ -15,7 +15,7 @@ package solvers
 import (
 	"fmt"
 
-	"github.com/che-incubator/che-workspace-operator/pkg/apis/workspace/v1alpha1"
+	"github.com/che-incubator/che-workspace-operator/pkg/apis/controller/v1alpha1"
```
Comment (sleshchenko) [inline pkg/controller/workspacerouting/solvers/openshift_oauth_solver.go:18]: Wow. Seems we need to rename module and replace all imports with new operator location )

File: pkg/controller/workspacerouting/workspacerouting_controller.go [hunk 2]
```diff
@@ -17,7 +17,7 @@ import (
 	"fmt"
 
 	"github.com/che-incubator/che-workspace-operator/internal/cluster"
-	workspacev1alpha1 "github.com/che-incubator/che-workspace-operator/pkg/apis/workspace/v1alpha1"
+	workspacev1alpha1 "github.com/che-incubator/che-workspace-operator/pkg/apis/controller/v1alpha1"
```
Comment (sleshchenko) [inline pkg/controller/workspacerouting/workspacerouting_controller.go:20]: I think the alias now should be renamed

Comment (sleshchenko) [reply pkg/controller/workspacerouting/workspacerouting_controller.go]: renamed to controllerv1alpha1

File: pkg/apis/controller/group.go [hunk 3]
```diff
@@ -0,0 +1,6 @@
+// Package controller contains controller API versions.
```
[the remark this answers is not in the document]
Comment (amisevsk) [reply pkg/apis/controller/group.go:13]: It should -- forgot to run `make fmt_license` but @sleshchenko caught it.

File: pkg/controller/workspacerouting/workspacerouting_controller.go (hunk 2 above)
Comment (amisevsk) [reply pkg/controller/workspacerouting/workspacerouting_controller.go]: :+1: Thank you

**Pull request merged**